src/main/java/edu/cornell/kfs/tax/batch/service/impl/TaxPayeeHelperServiceImpl.java

```diff
@@ -0,0 +1,190 @@
+            initializeVendorAddressData(payee, matchingVendor, statistics);
+        }
+    }
+
+    /*
+     * TODO: We need to revisit the parent-vendor-name setup and determine whether
+     * it should be renamed to something else.
+     */
+    private void initializeAlternateVendorNameIfSoleProprietor(final TaxPayeeBase payee,
+            final VendorDetailLite vendorDetail, final String taxType) {
+        final String soleProprietorOwnershipCode = getSuffixedParameter(
+                TaxCommonParameterNames.SOLE_PROPRIETOR_OWNER_CODE_PARAMETER_SUFFIX, taxType);
+        if (StringUtils.equals(vendorDetail.getVendorOwnershipCode(), soleProprietorOwnershipCode)) {
+            payee.setParentVendorName(vendorDetail.getVendorName());
+        }
+    }
+
+    private void initializeVendorName(final TaxPayeeBase payee,
+            final VendorDetailLite vendorDetail, final TaxStatisticsHandler statistics) {
+        final String vendorName = vendorDetail.getVendorName();
+        payee.setVendorName(vendorName);
+        if (StringUtils.isNotBlank(vendorName)) {
+            statistics.increment(TaxStatType.NUM_VENDOR_NAMES_PARSED);
+            if (vendorDetail.isVendorFirstLastNameIndicator()
+                    && StringUtils.contains(vendorName, KFSConstants.COMMA)) {
+                payee.setVendorLastName(StringUtils.trim(
+                        StringUtils.substringBefore(vendorName, KFSConstants.COMMA)));
+                payee.setVendorFirstName(StringUtils.trim(
+                        StringUtils.substringAfter(vendorName, KFSConstants.COMMA)));
+            } else {
+                payee.setVendorLastName(StringUtils.trim(vendorName));
+            }
+        } else {
+            statistics.increment(TaxStatType.NUM_VENDOR_NAMES_NOT_PARSED);
+        }
+    }
+
+    private void initializeVendorAddressData(final TaxPayeeBase payee,
+            final VendorDetailLite vendor, final TaxStatisticsHandler statistics) throws SQLException {
+        final VendorAddressLite usAddress = transactionDetailProcessorDao.getHighestPriorityUSVendorAddress(
+                vendor.getVendorHeaderGeneratedIdentifier(), vendor.getVendorDetailAssignedIdentifier());
+        final VendorAddressLite foreignAddress = transactionDetailProcessorDao.getHighestPriorityForeignVendorAddress(
+                vendor.getVendorHeaderGeneratedIdentifier(), vendor.getVendorDetailAssignedIdentifier());
+
+        if (ObjectUtils.isNotNull(usAddress)) {
+            payee.setVendorLine1Address(usAddress.getVendorLine1Address());
+            payee.setVendorLine2Address(usAddress.getVendorLine2Address());
+            payee.setVendorCityName(usAddress.getVendorCityName());
+            payee.setVendorStateCode(usAddress.getVendorStateCode());
+            payee.setVendorZipCode(usAddress.getVendorZipCode());
+            if (ObjectUtils.isNull(foreignAddress)) {
+                payee.setVendorEmailAddress(usAddress.getVendorAddressEmailAddress());
```

Comment:
why are we only setting the email address if the foreign address is null?

Reply:
For vendors that have both domestic and foreign addresses, we want the email from the foreign address to take precedence. However, since the logic in this method is already able to do that without the assistance of the is-null check above, I could remove the is-null check and just let the subsequent foreign address logic overwrite the email. Alternatively, I could look into having a separate block for setting the email address, to see if I can make the intent clearer.

Reply:
to avoid confusion, the alternative block might be the better approach